tests/inference/test_fitting.py

```
@@ -653,3 +661,451 @@ def nll(params):
+# _iminimize tests
+# ============================================================================
+
+
+class TestIminimize:
+    """Tests for _iminimize interactive minimization loop."""
+
+    def test_callback_called_each_iteration(self, simple_quadratic_nll, simple_params):
+        """Callback should be invoked at every solver step."""
+        call_count = []
+
+        def counting_callback(step, y, state):
```

Ruff (0.14.13)

Unused function argument: `y`

(ARG001)